types/time.go

```diff
@@ -1020,6 +1022,14 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int8, isFloat b
 
 			year, month, day, hour, minute, second =
 				dateTime.Year(), dateTime.Month(), dateTime.Day(), dateTime.Hour(), dateTime.Minute(), dateTime.Second()
+			if seps[0] == "0" {
```

Comment:
Add a test case for `sep[0] != "0"`

Reply:
added

Reply:
It seems that you have not added a test for `sep[0] != "0"` branch.

Reply:
you're right!  case `1.234` returned by `ParseDatetimeFromNum`.